test/services/SessionServiceSpec.scala

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import base.SpecBase
+import cats.data.EitherT
+import models.{InternalId, UserAnswers}
+import models.errors.SessionError
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.when
+import org.scalatest.matchers.should.Matchers.convertToAnyShouldWrapper
+import org.scalatestplus.mockito.MockitoSugar.mock
+import repositories.SessionRepository
+
+import scala.concurrent.{ExecutionContext, Future}
+class SessionServiceSpec extends SpecBase {
+
+  "Session Service" - {
+
+    implicit val ec: ExecutionContext = ExecutionContext.global;
+    val mockSessionRepository         = mock[SessionRepository]
+    val sessionService                = new SessionService(mockSessionRepository)
+    val id                            = InternalId("id")
+    val repositoryThrowable           = new Throwable("There was an error with sessionRepository")
+
+    "createUserAnswers should create answers if sessionRepository does not fail" in {
+      when(
+        mockSessionRepository.set(any())
+      ).thenReturn(
+        Future.successful(true)
+      )
+
+      val result: EitherT[Future, SessionError, Unit] = sessionService.createUserAnswers(id)
+      result.value.futureValue match {
```

Comment:
Might be nicer to rewrite this as 
result.value.futureValue mustBe Right() 
rather than having to map each value?

Although if the concern is more being explicit in the test, could wrap it in a clue

withClue("Session service should not fail to create answers"){
    result.value.futureValue mustBe Right() 
}

Reply:
Cool! I'll try that

Reply:
Yep that's better, I'll implement that